src/loaders.ts

```diff
@@ -178,11 +178,11 @@ type APIKeys = {
 
 // Example:
 // whatsabi.autoload(address, {provider, ...defaultsWithAPIKeys(process.env)})
-export function defaultsWithAPIKeys(apiKeys: APIKeys): Record<string, ABILoader|SignatureLookup> {
+export function defaultsWithAPIKeys(apiKeys: APIKeys, baseURL: string = 'https://api.etherscan.io/api'): Record<string, ABILoader|SignatureLookup> {
```

Comment:
Hm, what do you think of changing this to:

```typescript
type LoadersEnv = {
  ETHERSCAN_API_KEY?: string,
  ETHERSCAN_BASE_URL?: string,
  SOURCIFY_CHAIN_ID?: string,
}

export function defaultsWithEnv(env: LoadersEnv): Record<string, ABILoader|SignatureLookup> {
   ...
}
```

Can keep the old APIKeys stuff and I'll deprecate it, but this would have a superset of functionality, not just for keys.

Reply:
cc @jxom, I know you were using the loaders sourcify chain id override feature, would this be helpful? Or would a different API be better?

Reply:
Seems like we could actually just extend the APIKeys type:

```
type APIKeys = {
  ETHERSCAN_API_KEY?: string,
  ETHERSCAN_BASE_URL?: string,
  SOURCIFY_CHAIN_ID?: number,
}

// Example:
// whatsabi.autoload(address, {provider, ...defaultsWithAPIKeys(process.env)})
export function defaultsWithAPIKeys(apiKeys: APIKeys): Record<string, ABILoader|SignatureLookup> {
  return {
    abiLoader: new MultiABILoader([
      new SourcifyABILoader({chainId: apiKeys.SOURCIFY_CHAIN_ID}),
      new EtherscanABILoader({apiKey: apiKeys.ETHERSCAN_API_KEY, baseURL: apiKeys.ETHERSCAN_BASE_URL}),
    ]),
    signaturelookup: new MultiSignatureLookup([
      new OpenChainSignatureLookup(),
      new FourByteSignatureLookup(),
    ]),
  }
}
```